### PR TITLE
[wallet/desktop] task: add more component unit test

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- add more unit tests for component

## What's the fix?
- Added unit test for TransactionTypeSelector and TransactionUriDisplay
- Removed unused component WindowControls and TransactionListOptions
- Added missing transaction descriptor type `17229`, the description used same as Explorer

### CODE REVIEW

File: __tests__/components/TransactionUriDisplay.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        );
+
+        const component = wrapper.vm as TransactionUriDisplayTs;
+
+        transferTransaction.serialize();
+
+        jest.spyOn(TransactionURI.prototype, 'build').mockImplementation();
+
+        // Act:
+        component.transactionURI;
```
Comment (Jaguar0625) [inline __tests__/components/TransactionUriDisplay.spec.ts:36]: should you check returned value?

File: __tests__/components/TransactionTypeSelector.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    test('returns new value when user selected', () => {
+        // Arrange:
+        const wrapper = getTransactionTypeSelectorWrapper();
+        const component = wrapper.vm as TransactionTypeSelectorTs;
+
+        // Act:
+        component.chosenValue = 'ADDRESS_ALIAS';
+
+        // Assert:
+        expect(wrapper.emitted().input[0]).toEqual(['ADDRESS_ALIAS']);
```
Comment (Jaguar0625) [inline __tests__/components/TransactionTypeSelector.spec.ts:56]: also check property value (to cover getter)

Comment (AnthonyLaw) [reply __tests__/components/TransactionTypeSelector.spec.ts:56]: strange, getter is null, and trying to simulate a click on a selector is not working as expected.
will continue to debug on this

**Comment by yilmazbahadir** [comment]
@AnthonyLaw  can you pls rebase your changes on top of `feature/unit-tests-united`?
I think we shouldn't see highlighted commits in the image below.

image

**Comment by AnthonyLaw** [comment]
sure, I believe the PR is merged in the wrong branch.

**Comment by yilmazbahadir** [comment]
I fixed the merge issue, rebased `component-unit-test` and f-pushed. So you'll need to reset and pull the remote changes in your local.

File: __tests__/components/TransactionTypeSelector.spec.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    test('returns sorted transaction type list', () => {
+        // Arrange:
+        const wrapper = getTransactionTypeSelectorWrapper();
+        const component = wrapper.vm as TransactionTypeSelectorTs;
+
+        // Act:
+        const list = component.transactionTypeList;
+
+        // Assert:
+        expect(list).toEqual([
```
Comment (Jaguar0625) [inline __tests__/components/TransactionTypeSelector.spec.ts:19]: isn't ACCOUNT_OPERATION_RESTRICTION missing? 🤔

Comment (AnthonyLaw) [reply __tests__/components/TransactionTypeSelector.spec.ts:19]: `ACCOUNT_OPERATION_RESTRICTION` was filed out, not sure why 🤔

https://github.com/symbol/desktop-wallet/blob/aa834f28c77bd94d14194e3b25508bf00f41c876/src/components/TransactionTypeSelector/TransactionTypeSelectorTs.ts#L50

Comment (AnthonyLaw) [reply __tests__/components/TransactionTypeSelector.spec.ts:19]: found out the reason, this `transactionTypeList()` is used for ACCOUNT_OPERATION_RESTRICTION.
so you are able to block/allow all transaction types except itself (ACCOUNT_OPERATION_RESTRICTION)

Comment (Jaguar0625) [reply __tests__/components/TransactionTypeSelector.spec.ts:19]: it would be good to add a comment to that affect in the code 👍

**Review by Jaguar0625 — APPROVED** [review]

File: __tests__/components/TransactionTypeSelector.spec.ts (hunk 3 above)
Comment (AnthonyLaw) [reply __tests__/components/TransactionTypeSelector.spec.ts:19]: Sure thing, but Do you think I should rename to `OperationRestrictionTransactionTypeSelector` ?

**Comment by Jaguar0625** [comment]
you forgot to delete TransactionTypeSelectorTs.ts

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**